### ISSUE TITLE
Variable names like arr[0] is extracted as arr.undefined

### ISSUE THREAD

**Issue by MrOrz** [AlexMost/babel-plugin-polyglot#85]
Input (context: `tests/functional/test_extract_gettext_with_formatting.js`)

```
it('should extract gettext literal with varible names', () => {
  const input = 't`Size of the first pineapple: ${pineapples[0].length}`';
  babel.transform(input, options);
  const result = fs.readFileSync(output).toString();
  expect(result).to.contain('msgid "Size of the first pineapple: ${ pineapples[0].length }"');
});
```

Output:

```
AssertionError: expected 'msgid ""\nmsgstr ""\n"Content-Type: text/plain; charset=utf-8\\n"\n"Plural-Forms: nplurals=2; plural=(n!=1);\\n"\n\nmsgid "literal with formatting ${ a }"\nmsgstr ""\n\nmsgid "Size of the first pineapple: ${ pineapples.undefined.length }"\nmsgstr ""' to include 'msgid "Size of the first pineapple: ${ pineapples[0].length }"'
```

Expected result: the test should pass.

**Comment by AlexMost** [comment]
thanks for the report, recently fixed similar bug with `${ this.name }`

**Comment by AlexMost** [comment]
fixed - https://github.com/c-3po-org/babel-plugin-c-3po/pull/86
Will be available in `0.5.4` version.
